src/App.tsx

```diff
@@ -116,45 +117,72 @@ const doGetCharacterData = throttle(
     setBungieServiceUnavailable: (value: boolean) => void
   ) => {
     const updateCharacterData = (newData: PowerBarsCharacterData[]) => {
-      const newOverallPower = Math.max(...newData.map(c => c.overallPower));
+      const newOverallPower = Math.max(...newData.map((c) => c.overallPower));
       const newArtifactPower = Math.max(
-        ...newData.map(c => (c.artifactData ? c.artifactData.bonusPower : 0))
+        ...newData.map((c) => (c.artifactData ? c.artifactData.bonusPower : 0))
       );
       const newTotalPower = newOverallPower + newArtifactPower;
       ga.set({
         dimension1: `${newOverallPower}`,
         dimension2: `${newArtifactPower}`,
-        dimension3: `${newTotalPower}`
+        dimension3: `${newTotalPower}`,
       });
       setCharacterData(newData);
     };
 
     const isFetchingCharacterData = getIsFetchingCharacterData();
 
     (async () => {
-      if (!isFetchingCharacterData) {
-        try {
-          await getCharacterData(updateCharacterData);
-        } catch (error) {
-          if (error instanceof BungieSystemDisabledError) {
-            setBungieSystemDisabled(true);
-          }
-          if (error.message === "503") {
-            setBungieServiceUnavailable(true);
-          }
-          console.error("Error fetching character data:", error);
+      if (isFetchingCharacterData) return;
+
+      try {
+        await getCharacterData(updateCharacterData);
+      } catch (error) {
+        if (error instanceof BungieSystemDisabledError) {
+          setBungieSystemDisabled(true);
         }
+        if (error.message === "503") {
+          setBungieServiceUnavailable(true);
+        }
+        console.error("Error fetching character data:", error);
       }
```

Comment:
Little nit-pick, I inverted the if-statement here to remove the main code from an unnecessary level of indentation.